File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/commands/ConsoleAction.java
package io.github.tanguygab.conditionalactions.actions.types.commands;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.OfflinePlayer;

public class ConsoleAction extends Action {

    public ConsoleAction() {
        super("(?i)console:( )?");
    }

    @Override
    public String getSuggestion() {
        return "console: <command>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        String command = parsePlaceholders(player,match);
        getPlugin().getScheduler().global().run(()-> getPlugin().getServer().dispatchCommand(getPlugin().getServer().getConsoleSender(),command));
    }

}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/commands/ConsoleAction.kt
package io.github.tanguygab.conditionalactions.actions.types.commands

import io.github.tanguygab.conditionalactions.actions.Action
import org.bukkit.OfflinePlayer

class ConsoleAction : Action("^(?i)console:( )?".toRegex()) {
    override fun getSuggestion() = "console: <command>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val command = parsePlaceholders(player, match)
        plugin.sync(null) { plugin.server.dispatchCommand(plugin.server.consoleSender, command) }
    }
}
